front/src/modules/simulationResult/components/SpaceCurvesSlopes/SpaceCurvesSlopesV2.tsx

```diff
@@ -0,0 +1,274 @@
+      speed: speed * 3.6,
+      position: base.positions[i] / 1000,
+      time: base.times[i],
+    }));
+
+    // slopes
+    const gradients = slopes.map((step) => step.gradient);
+    const areaSlopesHistogram = mergeDatasAreaConstantV2(slopes, 0, ['position', 'gradient']);
+    const slopesCurve = createSlopeCurveV2(slopes, gradients);
+
+    // curves
+    const curvesHistogram = createCurvesCurveV2(curves, gradients);
+
+    return {
+      gradients,
+      speed: baseSpeedData,
+      slopesHistogram: slopes,
+      areaSlopesHistogram,
+      slopesCurve,
+      curvesHistogram,
+    };
+  }, [pathProperties]);
+
+  const timeScaleRange: [Date, Date] = useMemo(() => {
+    if (chart) {
+      const spaceScaleRange = chart.x.domain();
+      return spaceScaleRange.map((position) => interpolateOnPositionV2(trainData, position)) as [
+        Date,
+        Date,
+      ];
+    }
+    return [new Date(), new Date()];
+  }, [chart]);
+
+  const { updateTimePosition } = useChartSynchronizer(
+    (timePosition, positionValues) => {
+      if (dateIsInRange(timePosition, timeScaleRange)) {
+        traceVerticalLine(chart, CHART_AXES.SPACE_GRADIENT, positionValues, timePosition);
+      }
+    },
+    'space-curve',
+    [chart, timeScaleRange]
+  );
+
+  const createChart = (): SpeedSpaceChart => {
+    d3.select(`#${CHART_ID}`).remove();
+
+    const xMax = d3.max(trainData.slopesHistogram, (d) => d.position + 100) || 0;
+    const defineX = chart === undefined ? defineLinear(xMax) : chart.x;
+
+    const yMin = d3.min(trainData.slopesHistogram, (d) => d.gradient);
+    const yMax = d3.max(trainData.slopesHistogram, (d) => d.gradient) || 0;
+    const defineY = chart === undefined ? defineLinear(yMax, 0, yMin) : chart.y;
+
+    if (chart === undefined) {
+      setInitialYScaleDomain(defineY.domain());
+    }
+    let y2Min: number | undefined;
+    let y2Max: number | undefined;
+    let defineY2: d3.ScaleLinear<number, number, never> = defineLinear(0, 0, 0);
+
+    if (trainData.slopesCurve) {
+      y2Min = d3.min(trainData.slopesCurve, (d) => d.height);
+      y2Max = d3.max(trainData.slopesCurve, (d) => d.height) || 0;
+      defineY2 = chart === undefined ? defineLinear(y2Max, 0, y2Min) : chart.y2;
+    }
+
+    const width = parseInt(d3.select(`#container-${CHART_ID}`).style('width'), 10);
+    return defineChart(
+      width,
+      height,
+      defineX,
+      defineY,
+      ref as React.MutableRefObject<HTMLDivElement>,
+      false,
+      ['position', 'gradient'],
+      CHART_ID,
+      defineY2
+    ) as SpeedSpaceChart;
+  };
+
+  const drawOPs = (chartLocal: Chart) => {
+    const operationalPointsZone = chartLocal.drawZone
+      .append('g')
+      .attr('id', 'gev-operationalPointsZone');
+    operationalPoints!.forEach((op) => {
+      operationalPointsZone
+        .append('line')
+        .datum(op.position / 1000) // need to be in meters
```

Comment:
Use mmToM function l154, 163 and 169
```suggestion
        .datum(mmToM(op.position))
```

Reply:
Done